docs/source/developers/extensions.rst

```diff
@@ -306,6 +306,101 @@ To make your extension executable from anywhere on your system, point an entry-p
         }
     )
 
+Filtering requests
+------------------
+
+When launching the ``ExtensionApp`` in standalone mode, you may want to restrict the available endpoints. You can do this by defining
+either ``_allowed_spec`` or ``_blocked_spec`` class attributes as OpenAPI v3 specifications. The allowed paths will blocked any requests
+not matching the specification and the blocked paths will allow any requests except the ones specified.
+
+.. note::
+
+   If both are defined, the blocked paths take precedence on the allowed one; i.e. if a path is allowed and blocked, it will be blocked.
+
+OpenAPI v3 does not support URL path arguments that contains slashes ``/``. For the specification,
+``/api/contents/{path}`` can only be ``/api/contents/[^/]+``. To circumvent this limitation, you can provide a list of regex's through
+the class attributes ``_slash_encoder``. The groups matching one of the encoder will have their slashes encoded (``/`` replaced by ``%2F``).
+This will allow the validation of path through the OpenAPI v3 specification.
+
+Here is an example for the unit tests.
+
+.. code-block:: python
+
+    class MyExtensionApp(ExtensionApp):
+    
+        _allowed_spec = {
```

Comment:
It feels like these should be public attributes if they are meant to be part of the API

Reply:
Following the idea of jupyterlab/jupyterlab_server#167, the scenario imagined to design this PR is the following:

James wants to create a `SmithApp` that inherits from `ExtensionApp`. That app can be executed either as an Jupyter server extension or as a standalone application. But in the case of the standalone application, he only wants the new handlers of his extension and the get contents endpoint to be available.

To achieve this, James will overrides the protected class attribute `_allowed_spec` in the child class `SmithApp`. Users of the `SmithApp` will never be able to change the endpoints available in standalone mode*.

> *of course because this is Python, this is only a convention that protected attribute should not be modified externally.

pinging @bollwyvl to get his opinion